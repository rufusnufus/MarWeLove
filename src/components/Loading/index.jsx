import React from 'react'

import './index.css'

function Loading({ children, loading }) {
  if (loading) {
    return <img className="loading" src="/images/loading.gif" alt="loading" />
  }
  return <>{children}</>
}

export default Loading
