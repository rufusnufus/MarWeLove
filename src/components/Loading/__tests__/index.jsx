/**
 * @jest-environment jsdom
 */

import React from 'react'
import { cleanup, render } from '@testing-library/react'
import Loading from '../index'

afterEach(cleanup)

describe('Loading', () => {
it('Render off', () => {
  const { container } = render(
    <Loading loading={false} />
  )

  expect(container).toMatchSnapshot()
})

it('Render on', () => {
  const { container } = render(
    <Loading loading />
  )

  expect(container).toMatchSnapshot()
})
})
 