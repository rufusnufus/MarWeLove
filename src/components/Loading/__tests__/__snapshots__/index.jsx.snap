// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading Render off 1`] = `<div />`;

exports[`Loading Render on 1`] = `
<div>
  <img
    alt="loading"
    class="loading"
    src="/images/loading.gif"
  />
</div>
`;
